src/cargo/util/config/mod.rs

```diff
@@ -730,6 +743,14 @@ impl Config {
     /// Helper primarily for testing.
     pub fn set_env(&mut self, env: HashMap<String, String>) {
         self.env = env.into_iter().map(|(k, v)| (k.into(), v.into())).collect();
+        self.case_insensitive_env = self
+            .env_keys()
+            .map(|k| (k.to_uppercase(), k.to_owned()))
+            .collect();
+        self.normalized_env = self
+            .env()
+            .map(|(k, _)| (k.to_uppercase().replace("-", "_"), k.to_owned()))
+            .collect();
```

Comment:
I feel a little uncomfortable with all this duplication. Would it be possible to move this out to a free function that returns two HashMaps, and use that here and in the initializer?

Reply:
Oops, sorry this wasn't cleaner. I've added your suggestion in a new commit.